### PR TITLE
DIAN-548

### PR DESCRIPTION
**Pull request by dephillipsmichael**
@syoung-smallwisdom @nategbrown9 see Jira issue https://sagebionetworks.jira.com/browse/DIAN-548

Store and cancel jobs when refreshing callback connections. I'm hoping that cancelling the job specifically will allow iOS to call refreshStudyBurstSchedule and have it work as expected.

### CODE REVIEW

File: bridge-client/src/iosMain/kotlin/org/sagebionetworks/bridge/kmm/shared/NativeTimelineManager.kt [hunk 1]
```diff
@@ -86,7 +91,7 @@ class NativeTimelineManager(
     }
 
     fun refreshTodaySchedule() {
-        runCatching { scope.cancel() }
+        runCatching { todayJob?.cancel() }
```
Comment (nategbrown9) [inline bridge-client/src/iosMain/kotlin/org/sagebionetworks/bridge/kmm/shared/NativeTimelineManager.kt:95]: Could then set todayJob = null so cancel doesn't get called on it again.

File: bridge-client/src/iosMain/kotlin/org/sagebionetworks/bridge/kmm/shared/NativeTimelineManager.kt [hunk 2]
```diff
@@ -25,14 +26,16 @@ class NativeTimelineStudyBurstManager(
     private val updateFailed: (() -> Unit)?
 ) : AbstractNativeTimelineManager(studyId, scheduleMutator) {
 
+    var scheduleJob: Job? = null
+
     fun refreshStudyBurstSchedule(userJoinedDate: Instant) {
-        runCatching { scope.cancel() }
+        runCatching { scheduleJob?.cancel() }
```
Comment (nategbrown9) [inline bridge-client/src/iosMain/kotlin/org/sagebionetworks/bridge/kmm/shared/NativeTimelineManager.kt:32]: Could then set todayJob = null so cancel doesn't get called on it again.

**Review by nategbrown9 — APPROVED** [review]
Couple minor comments, otherwise looks good.

**Comment by dephillipsmichael** [comment]
@nategbrown9 I re-built the iOS library, and I also tested my changes locally, and they work!  I was able to refresh the schedule from bridge without having to force close the app.

**Review by syoung-smallwisdom — APPROVED** [review]
looks good

**Pull request merged**